[US-0011] - Implementa endpoint GET para contagem de posts promocionais

pierrecaladomeli opened the pull request:
A partir do ID de um vendedor

pierrecaladomeli committed:
[US-0011] - Implementa endpoint GET para contagem de posts promocionais de um vendedor

andreepdiasmeli commented on desafio_spring/src/main/java/meli/bootcamp/desafio_spring/controllers/PostController.java:
```diff
@@ -13,4 +19,18 @@
+
+
+    @GetMapping("/{userId}/countPromo")
+    public PromotionalCountDTO getPromotionalCountById(@PathVariable Long userId){
```
O ById do nome do método poderia ser SellerId para não confundir com postId, ou então remover esse ById do nome do método.

andreepdiasmeli commented on desafio_spring/src/main/java/meli/bootcamp/desafio_spring/services/UserService.java:
```diff
@@ -1,14 +1,27 @@
+    }
+
+    public PromotionalCountDTO getPromoProductsCount(Long sellerId){
+        Seller seller = sellerRepository.findById(sellerId).orElseThrow(() ->
```
Extrair essa linha para um método getSellerById, para não precisar duplicar código sempre que busccar um vendedor.

pierrecaladomeli committed:
[US-0011] - Refatora método getPromotionalCountById para getPromotionalCountBySellerId

andreepdiasmeli reviewed (approved):
topp

pierrecaladomeli committed:
[US-0011] - Resolve conflitos com a master

# Conflicts:
#	desafio_spring/src/main/java/meli/bootcamp/desafio_spring/controllers/PostController.java
#	desafio_spring/src/main/java/meli/bootcamp/desafio_spring/exceptions/ResourceNotFoundException.java
#	desafio_spring/src/main/java/meli/bootcamp/desafio_spring/repositories/SellerRepository.java
#	desafio_spring/src/main/java/meli/bootcamp/desafio_spring/services/PostService.java
#	desafio_spring/src/main/java/meli/bootcamp/desafio_spring/services/UserService.java
